### DISCUSSION TITLE
Feature request JWT

### DISCUSSION THREAD

**Discussion by at1993** [jchristn/WatsonWebsocket#156] [Ideas]
First of all, thanks @jchristn for this library as I've just ventured into this Websocket stuff.  Currently, the server uses IP filtering or let everyone in. it would be nice to let the application authenticate using JWT or some other type of token? In the WatsonWsClient class MessageReadAsync method, the input byte array seems to be not necessary as it is reassigned to a new buffer.

**Comment by jchristn** [comment]
Adding some (late) context here - the headers are passed in when a connection is established.  You could always set the JWT in the Authorization header, i.e. `Authorization: Bearer [token]`, and parse/process on connect.  The library wouldn't need to do anything in that case.

![Image](https://github.com/user-attachments/assets/91bbd697-2202-4670-bceb-b56efdeb5bf5)
